test/singletons.test.js

```diff
@@ -222,22 +223,54 @@ describe('singletons', function() {
   it('resolving singleton before first sync, then signing in, should result in correct number of instances', async function () {
     await this.registerUser();
     /** Create prefs and associate them with account */
-    const ogPrefs = await insertPrefsPayload(this.application);
+    const ogPrefs = await findOrCreatePrefsSingleton(this.application);
     await this.application.sync(syncOptions);
     this.application = await Factory.signOutApplicationAndReturnNew(this.application);
 
     /** Create another instance while signed out */
-    await insertPrefsPayload(this.application);
+    await findOrCreatePrefsSingleton(this.application);
     await Factory.loginToApplication({
       application: this.application,
       email: this.email,
       password: this.password
     });
     /** After signing in, the instance retrieved from the server should be the one kept */
-    const latestPrefs = await insertPrefsPayload(this.application);
+    const latestPrefs = await findOrCreatePrefsSingleton(this.application);
     expect(latestPrefs.uuid).to.equal(ogPrefs.uuid);
-    const allPrivs = this.application.itemManager.nonErroredItemsForContentType(ogPrefs.content_type);
-    expect(allPrivs.length).to.equal(1);
+    const allPrefs = this.application.itemManager.nonErroredItemsForContentType(ogPrefs.content_type);
+    expect(allPrefs.length).to.equal(1);
+  });
+
+  it('keeps local singleton when it is created before sync', async function () {
+    const identifier  = this.application.identifier;
+    await this.registerUser();
+
+    /** Create prefs and associate them with account */
+    const ogPrefs = await findOrCreatePrefsSingleton(this.application);
+    await this.application.sync(syncOptions);
+
+    /** Remove prefs locally and create a newer one */
+    await this.application.storageService.deletePayloads([ogPrefs]);
```

Comment:
Hmm, ogPrefs is an Item, but deletePayloads takes in a Payload, not an Item. You can probably use ogPrefs.payload.